pybamm/models/full_battery_models/lithium_ion/dfn.py

```diff
@@ -85,20 +85,38 @@ def set_particle_submodel(self):
             [particle_left, "Negative"],
             [particle_right, "Positive"],
         ]:
-            if particle_side == "Fickian diffusion":
-                self.submodels[
-                    domain.lower() + " particle"
-                ] = pybamm.particle.FickianManyParticles(self.param, domain)
-            elif particle_side in [
-                "uniform profile",
-                "quadratic profile",
-                "quartic profile",
-            ]:
-                self.submodels[
-                    domain.lower() + " particle"
-                ] = pybamm.particle.PolynomialManyParticles(
-                    self.param, domain, particle_side
-                )
+            if self.options["particle size"] == "single":
+                if particle_side == "Fickian diffusion":
+                    self.submodels[
+                        domain.lower() + " particle"
+                    ] = pybamm.particle.FickianManyParticles(self.param, domain)
```

Comment:
can we rename these submodels? the "many" here is confusing. maybe `XDistribution`/`XAveraged`?

Reply:
Yeah the whole naming convention with "many" and "single" is getting more cumbersome/confusing, I agree. Instead of making the class names longer, I've gone with submodules `size_distribution` and `no_distribution` (could do `size_averaged` instead) to differentiate them. Then the class names indicate just the transport model (Fickian, polynomial profile, etc) and whether it is "x-averaged". For example,

`particle.FickianManyParticles` is now `particle.no_distribution.FickianDiffusion`
`particle.FickianSingleParticles` is now `particle.no_distribution.XAveragedFickianDiffusion`
`particle.FickianManySizeDistributions` is now `particle.size_distribution.FickianDiffusion`
`particle.FickianSingleSizeDistribution` is now `particle.size_distribution.XAveragedFickianDiffusion`

Could also maybe group them into `x_averaged` and `x_full` submodules in a similar way, if that would make it even clearer. (Easily changed)